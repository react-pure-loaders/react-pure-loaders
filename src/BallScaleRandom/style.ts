import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const ballBeat = keyframes(
  {
    '100%': {
      opacity: 1,
      transform: 'scale(1)',
    },
    '50%': {
      opacity: 0.2,
      transform: 'scale(0.75)',
    },
  },
);

export default (color?: string) => ({
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    '&:nth-of-type(2n-1)': {
      animationDelay: '-0.35s',
    },

    animation: `${ballBeat} 0.7s 0s infinite linear`,
    backgroundColor: color || PRIMARY_COLOR,
    display: 'inline-block',
  },
});
