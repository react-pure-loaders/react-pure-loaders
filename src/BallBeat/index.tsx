/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
