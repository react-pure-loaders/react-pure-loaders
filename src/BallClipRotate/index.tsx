/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
