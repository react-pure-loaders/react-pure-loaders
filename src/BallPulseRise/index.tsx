/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
