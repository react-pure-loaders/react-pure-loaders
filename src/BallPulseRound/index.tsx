/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
