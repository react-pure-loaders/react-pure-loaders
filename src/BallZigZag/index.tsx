/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
