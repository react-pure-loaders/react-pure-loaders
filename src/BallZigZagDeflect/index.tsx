/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
