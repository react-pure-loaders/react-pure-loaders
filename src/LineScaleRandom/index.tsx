/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
