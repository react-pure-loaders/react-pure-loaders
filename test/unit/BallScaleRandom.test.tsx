import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallScaleRandom from '../../src/BallScaleRandom';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallScaleRandom>', () => {
    afterEach(cleanup);

    test('BallScaleRandom should match snapshot', () => {
        const { container } = render(<BallScaleRandom loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallScaleRandom should have default color', () => {
        const { container } = render(<BallScaleRandom loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('BallScaleRandom should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallScaleRandom color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallScaleRandom should have no children', () => {
        const { container } = render(<BallScaleRandom loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallScaleRandom should have three children', () => {
        const { container } = render(<BallScaleRandom loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
