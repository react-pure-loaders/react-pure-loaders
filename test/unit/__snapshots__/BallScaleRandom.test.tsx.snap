// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallScaleRandom> BallScaleRandom should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.2;
    -webkit-transform: scale(0.75);
    -moz-transform: scale(0.75);
    -ms-transform: scale(0.75);
    transform: scale(0.75);
  }
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 0.7s 0s infinite linear;
  animation: animation-0 0.7s 0s infinite linear;
  display: inline-block;
}

.emotion-0>div:nth-of-type(2n-1) {
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
